internal/controllers/import_controller.go

```diff
@@ -66,10 +64,12 @@ func (r *CAPIImportReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Ma
 		r.remoteClientGetter = remote.NewClusterClient
 	}
 
-	// TODO: we want the control plane initialized but removing for the time being
-	// capiPredicates := predicates.All(log, predicates.ClusterControlPlaneInitialized(log), predicates.ResourceHasFilterLabel(log, r.WatchFilterValue))
-	capiPredicates := predicates.All(log, predicates.ResourceHasFilterLabel(log, r.WatchFilterValue),
-		turtlespredicates.ClusterWithoutImportedAnnotation(log))
+	capiPredicates := predicates.All(log,
+		predicates.ResourceHasFilterLabel(log, r.WatchFilterValue),
+		turtlespredicates.ClusterWithoutImportedAnnotation(log),
+		turtlespredicates.ClusterWithReadyControlPlane(log),
+		turtlespredicates.ClusterOrNamespaceWithImportLabel(ctx, log, r.Client, importLabelName),
```

Comment:
This will no longer work. Namespace is a different object, and predicates.All is an equivalent of && here. Each of the previous predicates are checking if the object is a Cluster object.

Reply:
This also checks if its a Cluster object .

Reply:
+1

Reply:
it works as expected, the predicate checks if object is a cluster and gets namespace if label is not found